src/sagemaker/workflow/pipeline.py

```diff
@@ -622,6 +707,47 @@ def is_cyclic_helper(current_step):
                     return True
         return False
 
+    def build_adjacency_list_with_condition_edges(self) -> List[Dict[str, any]]:
+        """Generates an adjacency list that includes edge labels for Condition Steps"""
```

Comment:
The adjacency list is fixed once a pipeline graph object is created. Please follow the existing code which
- call `_initialize_adjacency_list` in `__init__` and assign the return value to an instance object `self.adjacency_list`
- Then we don't need to recreate the adjacency list every time we call .display